Code/Framework/AzFramework/Tests/QualitySystemComponentTests.cpp

```diff
@@ -0,0 +1,242 @@
+                }
+            }
+            )");
+
+        // expect the cvars are created with their default values
+        auto value = AzFramework::QualityLevel::LevelFromDeviceRules;
+        EXPECT_EQ(m_console->GetCvarValue("q_test", value), AZ::GetValueResult::Success);
+        EXPECT_EQ(value, AzFramework::QualityLevel{1});
+
+        EXPECT_EQ(m_console->GetCvarValue("q_test_sub", value), AZ::GetValueResult::Success);
+        EXPECT_EQ(value, AzFramework::QualityLevel::DefaultQualityLevel);
+    }
+
+    AZ_CVAR(int32_t, a_setting, 0, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Example integer setting 1");
+    AZ_CVAR(AZ::CVarFixedString, b_setting, "default", nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Example string setting 2");
+    AZ_CVAR(int32_t, c_setting, -1, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Example integer setting 3");
+    AZ_CVAR(int32_t, d_setting, -2, nullptr, AZ::ConsoleFunctorFlags::DontReplicate, "Example integer setting 4");
+
+    TEST_F(QualitySystemComponentTestFixture, QualitySystem_Loads_Group_Level)
+    {
+        // when the quality system component registers group cvars
+        StartApplicationWithSettings(R"(
+            {
+                "O3DE": {
+                    "Quality": {
+                        "DefaultGroup":"q_test",
+                        "Groups": {
+                            "q_test": {
+                                "Levels": [ "low", "medium","high", "veryhigh"],
+                                "Default": 2,
+                                "Description": "q_test quality group",
+                                "Settings": {
+                                    "a_setting": [0,1,2,3],
+                                    "b_setting": ["a","b","c","d"],
+                                    "q_test_sub": [0,1,1,1]
+                                }
+                            },
+                            "q_test_sub": {
+                                "Levels": [ "low", "high" ],
+                                "Default": "DefaultQualityLevel",
+                                "Description": "q_test_sub quality group",
+                                "Settings": {
+                                    "c_setting": [123,234],
+                                    "d_setting": [42] // test missing high level setting
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            )");
+
+        auto value = AzFramework::QualityLevel::LevelFromDeviceRules;
+        int32_t intValue = -42;
+        AZ::CVarFixedString stringValue;
+
+        // expect the value  defaults
+        EXPECT_EQ(m_console->GetCvarValue("a_setting", intValue), AZ::GetValueResult::Success);
+        EXPECT_EQ(intValue, 0);
+        EXPECT_EQ(m_console->GetCvarValue("b_setting", stringValue), AZ::GetValueResult::Success);
+        EXPECT_STREQ(stringValue.c_str(), "default");
```

Comment:
FYI we have added pretty printers for the string types in googletest https://github.com/o3de/o3de/blob/ef63f6fa8ffb30027a7bdc3f4d349766531c6d19/Code/Framework/AzTest/AzTest/Printers.h#L16-L30, 

So EXPECT_EQ can be used directly.
```suggestion
        EXPECT_EQ("default", stringValue);
```